src/Core/Entities/OrganizationDomain.cs

```diff
@@ -11,8 +11,33 @@ public class OrganizationDomain : ITableObject<Guid>
     [MaxLength(255)]
     public string DomainName { get; set; }
     public DateTime CreationDate { get; set; } = DateTime.UtcNow;
-    public DateTime? VerifiedDate { get; set; }
-    public DateTime NextRunDate { get; set; }
-    public int NextRunCount { get; set; }
+    public DateTime? VerifiedDate { get; private set; }
+    public DateTime NextRunDate { get; private set; }
+    public int NextRunCount { get; private set; }
     public void SetNewId() => Id = CoreHelpers.GenerateComb();
+
+    public OrganizationDomain SetNextRunDate()
+    {
+        var count = NextRunCount == 0 ? NextRunCount++ : NextRunCount;
+        
+        NextRunDate = CreationDate.AddHours(count * 12);
+        return this;
+    }
+
+    public OrganizationDomain SetNextRunCount()
+    {
+        if (NextRunCount == 3)
+        {
+            return this;
+        }
+
+        NextRunCount++;
+        return this;
+    }
+
+    public OrganizationDomain SetVerifiedDate()
+    {
+        VerifiedDate = DateTime.UtcNow;
+        return this;
```

Comment:
Is it necessary to return the instances of the OrganizationDomain in these calls? I'm not seeing this as a pattern in the other Entities.

Reply:
> Is it necessary to return the instances of the OrganizationDomain in these calls? I'm not seeing this as a pattern in the other Entities.

I used this to allow chaining, but it can be removed to return `void`

Reply:
Understood; you can leave it if it is useful! I don't see these methods being used this way though.